lib/integrations/awsoidc/deployservice_update.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsoidc
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ecs"
+	ecsTypes "github.com/aws/aws-sdk-go-v2/service/ecs/types"
+	"github.com/aws/aws-sdk-go/aws/awsutil"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/modules"
+)
+
+// waitDuration specifies the amount of time to wait for a service to become healthy after an update.
+const waitDuration = time.Minute * 5
+
+// UpdateDeployServiceAgents updates the deploy service agents with the specified teleportVersionTag.
+func UpdateDeployServiceAgents(ctx context.Context, clt DeployServiceClient, teleportClusterName, teleportVersionTag string, ownershipTags AWSTags) error {
```

Comment:
Would it make sense to make this (and `updateDeployServiceAgent` below) a member of `DeployServiceClient`? E.g. `clt.UpdateAgents()`.

Reply:
I don't think `UpdateDeployServiceAgent` should be a member of the interface, but I'll add `UpdateServiceRequest` struct to simplify the function def.